python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -946,7 +965,12 @@ def inner(fn: Callable[..., Any]) -> AssetsDefinition:
             if asset_in.partition_mapping
         }
 
-        named_ins = build_named_ins(fn, ins or {}, set())
+        upstream_asset_deps = make_asset_deps(deps=deps)
+        named_ins = build_named_ins(
+            fn=fn,
+            asset_ins=ins or {},
+            deps=({dep.asset_key for dep in upstream_asset_deps} if upstream_asset_deps else set()),
+        )
```

Comment:
Cannot comment directly on the line in question, but on this line https://github.com/dagster-io/dagster/pull/23804/files#diff-0fcad2d82f20f3987322def0a04db6e04bc81b9777e5ced50f0fed5e527872bdR995

We convert all `named_ins` (`ins` and `deps`) to `GraphIns`. `GraphIns` do not hold type information (pr - https://github.com/dagster-io/dagster/pull/4876, discussion - https://github.com/dagster-io/dagster/pull/4624#pullrequestreview-740582469) 

The `GraphIns` are later converted to `InputDefinitions` and those input defs are what is processed in `do_composition`. So even with the param flip in https://github.com/dagster-io/dagster/pull/23804/files#r1725750404 using `deps` still causes errors because we do not have the type information to detect that a `Nothing` type has been passed and can be ignored as a compute function parameter